### PR TITLE
vector: 0.25.1 -> 0.25.2

### PR DESCRIPTION
**Pull request by ghuntley**
###### Description of changes

nb: updated from GitHub web editor -> going to wait for build to fail to learn the correct hash then update 📦 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/vector/default.nix [hunk 1]
```diff
@@ -44,7 +44,7 @@ rustPlatform.buildRustPackage {
     hash = "sha256-7iYiSO966o0M9M0ijGCpuRVRgus+tURLBN9S5lPDRb8=";
```
Comment (ghuntley) [inline pkgs/tools/misc/vector/default.nix:44]: This hash needs to be updated.

**Comment by SuperSandro2000** [comment]
> nb: updated from GitHub web editor -> going to wait for build to fail to learn the correct hash then update package

This means you cannot have tested this update at all.

**Comment by ghuntley** [comment]
> This means you cannot have tested this update at all.

Correct. The package is backed by integration tests however...

**Review by happysalada — CHANGES_REQUESTED** [review]
The github hash is not updated it seems

File: pkgs/tools/misc/vector/default.nix (hunk 1 above)
Comment (ghuntley) [reply pkgs/tools/misc/vector/default.nix:44]:
```
nix-prefetch-git --url https://github.com/vectordotdev/vector  --rev v0.25.2
Initialized empty Git repository in /tmp/git-checkout-tmp-OOxCoXBD/vector/.git/
remote: Enumerating objects: 7770, done.
remote: Counting objects: 100% (7770/7770), done.
remote: Compressing objects: 100% (5832/5832), done.
remote: Total 7770 (delta 558), reused 5614 (delta 448), pack-reused 0
Receiving objects: 100% (7770/7770), 16.10 MiB | 12.07 MiB/s, done.
Resolving deltas: 100% (558/558), done.
From https://github.com/vectordotdev/vector
 * tag               v0.25.2    -> FETCH_HEAD
Switched to a new branch 'fetchgit'
removing `.git'...

git revision is d690c440ded1646e07d3c7dee73db09953f745bb
path is /nix/store/ws0h32b5q0nl4s6kn1ms67fr9ng3lysn-vector
git human-readable version is -- none --
Commit date is 2022-11-29 09:46:18 -0800
hash is 0c0bbqp6k4nirw347r54sp9mpsfyzyvb7v47rrs6wmy0nxlmaj42
{
  "url": "https://github.com/vectordotdev/vector",
  "rev": "d690c440ded1646e07d3c7dee73db09953f745bb",
  "date": "2022-11-29T09:46:18-08:00",
  "path": "/nix/store/ws0h32b5q0nl4s6kn1ms67fr9ng3lysn-vector",
  "sha256": "0c0bbqp6k4nirw347r54sp9mpsfyzyvb7v47rrs6wmy0nxlmaj42",
  "fetchLFS": false,
  "fetchSubmodules": false,
  "deepClone": false,
  "leaveDotGit": false
}
```

**Comment by happysalada** [comment]
@GrahamcOfBorg eval

**Comment by happysalada** [comment]
The evaluation fails for some reason, trying to restart it

**Comment by happysalada** [comment]
It says that the hash is invalid 

Have you ever seen the tool nix-update ? Thats what i use to make updates usually

**Comment by happysalada** [comment]
Hey i ended up doing this in https://github.com/NixOS/nixpkgs/pull/204361

**Pull request closed without merging**